neat_stat.c

```diff
@@ -0,0 +1,36 @@
+#include <stdint.h>
+#include <stdio.h>
+#include <stdarg.h>
+#include <stdlib.h>
+#include <string.h>
+
+#include "neat_internal.h"
+#include "neat_core.h"
+#include "neat_stat.h"
+
+
+/* Traverse the relevant subsystems of NEAT and gather the stats
+   then format the stats as a json string to return */	
+void neat_stats_build_json(neat_flow *flow, char **json_stats)
+{
+	neat_log(NEAT_LOG_DEBUG, "%s", __func__);
+
+	json_t *root = json_object();
```

Comment:
I think we reached concensus that we should avoid declaring variables within blocks, maybe move declaration to top?